assembl/graphql/schema.py

```diff
@@ -331,26 +332,26 @@ def _maybe_translate(post, locale, request):
         if request.authenticated_userid == Everyone:
             # anonymous cannot trigger translations
             return
-        lpc = LanguagePreferenceCollection.getCurrent(request)
+        if locale:
+            lpc = LanguagePreferenceCollectionWithDefault(locale)
+        else:
+            lpc = LanguagePreferenceCollection.getCurrent(request)
         for ls in (post.body, post.subject):
             source_locale = ls.first_original().locale_code
-            if locale:
-                target_locale = locale
-            else:
-                pref = lpc.find_locale(source_locale)
-                target_locale = pref.translate_to_locale
-                if not target_locale:
-                    continue
-                target_locale = target_locale.code
+            pref = lpc.find_locale(source_locale)
+            target_locale = pref.translate_to_locale
+            if not target_locale:
+                continue
+            target_locale = target_locale.code
             if not ls.closest_entry(target_locale):
                 post.maybe_translate(lpc)
 
     def resolve_subject_entries(self, args, context, info):
-        lp = LanguagePreferenceCollection.getCurrent()
-        lang = lp.default_locale_code()
+        # Use self.subject and not self.get_subject() because we still
```

Comment:
`get_subject` pops the message from the object? If so, the method name is misleading.

Reply:
get_subject returns EMPTY langstring for deleted post. We want this behavior in v1, but not in v2...